src/org/jetbrains/java/decompiler/modules/decompiler/exps/ConstExprent.java

```diff
@@ -353,13 +353,13 @@ else if (UNINLINED_FLOATS.containsKey(floatVal)) {
       // Check for special values that can't be used directly in code
       // (and we can't replace with the constant due to the user requesting not to)
       if (Float.isNaN(floatVal)) {
-        return new TextBuffer("0.0F / 0.0F");
+        return new TextBuffer("0.0F / 0.0");
       }
       else if (floatVal == Float.POSITIVE_INFINITY) {
-        return new TextBuffer("1.0F / 0.0F");
+        return new TextBuffer("1.0F / 0.0");
       }
       else if (floatVal == Float.NEGATIVE_INFINITY) {
-        return new TextBuffer("-1.0F / 0.0F");
+        return new TextBuffer("-1.0F / 0.0");
       }
```

Comment:
These changes should be reverted.

https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/Float.java#L70-L89